server/src/QRServer/db/connector.py

```diff
@@ -56,25 +57,66 @@ async def get_user_by_username(self, username) -> Optional[DbUser]:
             user_id=row[0],
             username=row[1],
             password=row[2],
-            created_at=row[3]
+            created_at=row[3],
+            discord_user_id=row[4],
         )
 
-    async def authenticate_user(self, username: str, password: Optional[bytes], auto_create=False,
-                                verify_password=True) -> Optional[DbUser]:
+    async def get_users_by_discord_id(self, discord_user_id) -> List[DbUser]:
+        c = await self.conn.cursor()
+        await c.execute(
+            "select id, username, password, created_at, discord_user_id from users where discord_user_id = ?", (
+                discord_user_id,
+            ))
+        rows = await c.fetchall()
+        if rows is None:
+            return []
+        result = []
+
+        for row in rows:
+            result.append(
+                DbUser(
+                    user_id=row[0],
+                    username=row[1],
+                    password=row[2],
+                    created_at=row[3],
+                    discord_user_id=row[4]
+                )
+            )
+        return result
+
+    async def create_user(self, username: str, password: bytes, discord_user_id: Optional[str] = None) -> DbUser:
+        c = await self.conn.cursor()
+        await c.execute(
+            "insert into users("
+            "  id, username, password, created_at, discord_user_id"
+            ") values (?, ?, ?, ?, ?)", (
+                str(uuid.uuid4()),
+                username,
+                password_hash(password) if password else None,
+                datetime.now().timestamp(),
+                discord_user_id,
+            ))
+        await self.conn.commit()
+        return await self.get_user_by_username(username)
+
+    async def authenticate_user(self, username: str, password: Optional[bytes], discord_user_id: Optional[str] = None,
```

Comment:
Why do we need `discord_user_id` in `authenticate_user`?

Reply:
we don't, removed